user/user-util/src/java/org/sakaiproject/user/util/ConvertUserFavoriteSitesSakai11.java

```diff
@@ -0,0 +1,554 @@
+            info("You must set the tomcat.dir system property.\n");
+            showUsage();
+
+            System.exit(1);
+        }
+
+        try {
+            Properties sakaiProperties = loadSakaiProperties(tomcatDir + File.separator + "sakai" + File.separator + "sakai.properties");
+
+            Class.forName(getProperty(sakaiProperties, "driverClassName@javax.sql.BaseDataSource"));
+
+            String jdbcUrl = getProperty(sakaiProperties, "url@javax.sql.BaseDataSource");
+
+            Connection db = null;
+            try {
+                db = DriverManager.getConnection(jdbcUrl,
+                                                 getProperty(sakaiProperties, "username@javax.sql.BaseDataSource"),
+                                                 getProperty(sakaiProperties, "password@javax.sql.BaseDataSource"));
+
+                migrateFavoriteSites(db);
+            } finally  {
+                db.close();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void migrateFavoriteSites(Connection db) throws
+        SQLException {
+        PreparedStatement countPreferences = null;
+        PreparedStatement usersSiteMemberships = null;
+        PreparedStatement selectPreferences = null;
+        PreparedStatement updatePreference = null;
+        ResultSet rs = null;
+
+        db.setAutoCommit(false);
+
+        try {
+            countPreferences = db.prepareStatement("select count(1) from sakai_preferences");
+
+            rs = countPreferences.executeQuery();
+            int totalPreferences = 0;
+            if (rs.next()) {
+                totalPreferences = rs.getInt(1);
+            }
+
+            info("Preferences to migrate: " + totalPreferences);
+
+            ProgressCounter counter = new ProgressCounter(totalPreferences);
+
+            // There's a lot here!  What we're doing:
+            //
+            //  * Find any user who has a preferences entry
+            //
+            //  * Join them with their site memberships, where those sites:
+            //
+            //      - are published (or where the user has access to the unpublished site)
+            //      - are not a workspace
+            //      - are not a special site
+            //      - haven't been deleted
+            //
+            //  * Sites that don't meet this criteria will yield NULLs, and
+            //    that's OK--this ensures we always get at least one row for a
+            //    given preference record, even if that user isn't in any sites.
+            //    We'll drop them out as we notice them.
+            usersSiteMemberships = db.prepareStatement("select p.preferences_id as userId, ss.site_id as siteId from" +
+                                                       " sakai_preferences p" +
+                                                       " left outer join sakai_site_user ssu on ssu.user_id = p.preferences_id" +
```

Comment:
SAKAI_SITE_USER should be capitalized. Same with SAKAI_PREFERENCES and SAKAI_SITE